hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/CatalogReplicaReplicationEndpoint.java

```diff
@@ -0,0 +1,493 @@
+        } else {
+          // Check if it can be merged into the last set.
+          List<Entry> l = ll.get(ll.size() - 1);
+          WALProtos.FlushDescriptor flushDesc =
+            WALEdit.getFlushDescriptor(l.get(0).getEdit().getCells().get(0));
+          if (flushDesc != null) {
+            // Flush events need to be in their own list
+            List<Entry> newList = new ArrayList<>();
+            newList.add(entry);
+            ll.add(newList);
+          } else {
+            // Check if the entry itself is a flush event.
+            flushDesc = WALEdit.getFlushDescriptor(edit.getCells().get(0));
+            if (flushDesc != null) {
+              // Flush events need to be in their own list
+              List<Entry> newList = new ArrayList<>();
+              newList.add(entry);
+              ll.add(newList);
+            } else {
+              l.add(entry);
+            }
+          }
+        }
+      } catch (IOException ioe) {
+        LOG.warn("Failed to parse {}", entry.getEdit().getCells().get(0));
+        skippedEdits++;
+      }
+    }
+
+    // queue edits to shippers.
+
+    for (Map.Entry<String, List<List<Entry>>> entry : encodedRegionName2Entries.entrySet()) {
+      CatalogReplicaShipper[] replicaShippers = primaryRegionsUnderReplication.get(entry.getKey());
+      if (replicaShippers != null) {
+        for (List<Entry> l : entry.getValue()) {
+          for (CatalogReplicaShipper shipper : replicaShippers) {
+            shipper.addEditEntries(l);
+          }
+        }
+      } else {
+        skippedEdits ++;
+      }
+    }
+    ctx.getMetrics().incrLogEditsFiltered(skippedEdits);
+
+    return true;
+  }
+
+  @Override public UUID getPeerUUID() {
+    return this.ctx.getClusterId();
+  }
+
+  @Override public boolean canReplicateToSameCluster() {
+    return true;
+  }
+
+  @Override protected WALEntryFilter getScopeWALEntryFilter() {
+    // we do not care about scope. We replicate everything.
+    return null;
+  }
+
+  @Override public void start() {
+    startAsync();
+  }
+
+  @Override public void stop() {
+    // Stop shippers
+    for (Map.Entry<String, CatalogReplicaShipper[]> e : primaryRegionsUnderReplication.entrySet()) {
+      for (CatalogReplicaShipper s : e.getValue()) {
+        s.setRunning(false);
+      }
+    }
+    stopAsync();
+  }
+
+  @Override protected void doStart() {
+    notifyStarted();
+  }
+
+  @Override protected void doStop() {
+    notifyStopped();
+  }
+
+  public int getNumRetries() {
+    return numRetries;
+  }
+
+  public long getOperationTimeoutNs() {
+    return operationTimeoutNs;
+  }
+
+  private static class CatalogReplicaShipper extends Thread {
+    private final CatalogReplicaReplicationEndpoint endpoint;
+    private LinkedBlockingQueue<List<Entry>> queue;
+    private final AsyncClusterConnection conn;
+    //Indicates whether this particular worker is running
+    private volatile boolean isRunning = true;
```

Comment:
isRunning is name of method to access a boolean. Change this data member name to be 'running'.